lambdas/build-stac/utils/stac.py

```diff
@@ -83,8 +91,6 @@ def generate_stac_regexevent(item: events.RegexEvent) -> pystac.Item:
         properties["start_datetime"] = start_datetime.isoformat()
         properties["end_datetime"] = end_datetime.isoformat()
         single_datetime = None
-    else:
-        single_datetime = single_datetime.isoformat()
```

Comment:
Why was this removed? My understanding is that `single_datetime` will be a `datetime` object unless we either null it out (as we do on line 85) or convert it to ISO string format (as we were doing on line 87)

https://github.com/NASA-IMPACT/veda-data-pipelines/blob/d713ff9f90040da26c54c450ef1375ef583a991a/lambdas/build-stac/utils/regex.py#L30-L32
https://github.com/NASA-IMPACT/veda-data-pipelines/blob/d713ff9f90040da26c54c450ef1375ef583a991a/lambdas/build-stac/utils/events.py#L19-L21

extract_dates()

Reply:
apparently the `create_item` call needs the datetime value to be a datetime object, not a string

https://github.com/developmentseed/rio-stac/blob/master/rio_stac/stac.py#L239